packages/cloudflare/src/utils/non-server-chunk-detector.ts

```diff
@@ -15,6 +15,8 @@ export class NonServerChunkDetector {
 		return {
 			name: 'non-server-chunk-detector',
 			generateBundle: (_, bundle) => {
+				// Skip if we bundle for client
+				if (!bundle['index.js']) return;
```

Comment:
We could just not add the plugin for the client run, right?

Reply:
Well how would you do that? The client and server run use the same `vite` object 🤔

Reply:
AFAIK  the object reference is the same. So if you add it in for the `server` target it still exists for the `client` target, and you would actually have to remove it manually again, see: https://github.com/withastro/adapters/blob/1927f941b41ac7156054c337edb031be8132e3e2/packages/cloudflare/src/index.ts#L268-L274

Reply:
Interesting...
I remember testing something related to this and it worked. Maybe it is only a shallow clone and nested objects are shared.

Well, if it doesn' work in this case then how you did LGTM :)